src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -1033,14 +1033,14 @@ public Builder withProxy(String proxyHost, int proxyPort) {
          * @return A new {@link Pinecone} instance configured based on the builder parameters.
          */
         public Pinecone build() {
-            PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig);
+            PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig, customOkHttpClient);
```

Comment:
`customOkHttpClient` being passed to the `PineconeConfig` constructor at the time of initializing `PineconeConfig` in `Pinecone` class